### PR TITLE
feat(developer): LDML keyboard test window marker support 🌱

### LINKED ISSUE #10416: refactor(developer): Use `km_core_state_action_items` struct instead of queue
See #9720 for details.

### LINKED ISSUE #10458: bug(developer): LDML keyboard debugger does not show markers
We need to query the context directly in order to retrieve markers, as they will not be emitted in actions. This requires a bit of a refactor to the debug form, to use the new action struct, and then splice in markers with get_context.

This same work really could be applied to the KMN keyboard debugger, but it's a lot more complicated due to the interleaving of actions and debug events.

Relates to #10416.

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #10458.
Fixes #10416.

Moves from using action queue to action struct for the LDML keyboard debugger. Does not do the same for the KMX keyboard debugger, because that relies on action items to do single-step debugging.

Refactors the LDML keyboard test window significantly, in order to be able to support markers. Cleans up the state machine for the debug window in the process, as that was complicating the changes. Removes a lot of extraneous code in the debug window.

# User Testing

* **TEST_LDML_DEBUGGER:** Create a test LDML keyboard, open it, and paste the sample keyboard XML from below into the text editor. Select Keyboard/Test Keyboard. Try the following interactions with the keyboard debugger and make sure it works as expected:

1. basic output: type <kbd>x</kbd>, output should be `hello!`
2. marker output: type <kbd>y</kbd>, output should be `\m{3}` in the character preview
3. basic transform: type <kbd>a</kbd><kbd>b</kbd><kbd>c</kbd>, output should be `Congratulations!`
4. marker transform: type <kbd>y</kbd><kbd>z</kbd>, output should be `b`, `\m{2}` in the character preview

### sample keyboard

```xml
<?xml version="1.0"?>
<!DOCTYPE keyboard3 SYSTEM "ldmlKeyboard3.dtd">
<keyboard3 locale="en" conformsTo="techpreview">
  <info author="Marc" name="sample_ldml"/>
  <version number="1.0.0"/>
  <keys>
    <import base="cldr" path="techpreview/keys-Zyyy-punctuation.xml"/>
    <import base="cldr" path="techpreview/keys-Zyyy-currency.xml"/>
    <key id="x" output="hello!" />
    <key id="y" output="\m{y}" />
  </keys>
  <layers formId="us">
    <layer modifiers="none">
      <row keys="grave 1 2 3 4 5 6 7 8 9 0 hyphen equal"/>
      <row keys="q w e r t y u i o p open-square close-square backslash"/>
      <row keys="a s d f g h j k l semi-colon apos"/>
      <row keys="z x c v b n m comma period slash"/>
      <row keys="space"/>
    </layer>
    <layer modifiers="shift">
      <row keys="tilde bang at hash dollar percent caret amp asterisk open-paren close-paren underscore plus"/>
      <row keys="Q W E R T Y U I O P open-curly close-curly pipe"/>
      <row keys="A S D F G H J K L colon double-quote"/>
      <row keys="Z X C V B N M open-angle close-angle question"/>
      <row keys="space"/>
    </layer>
    <layer modifiers="caps">
      <row keys="grave 1 2 3 4 5 6 7 8 9 0 hyphen equal"/>
      <row keys="Q W E R T Y U I O P open-square close-square backslash"/>
      <row keys="A S D F G H J K L semi-colon apos"/>
      <row keys="Z X C V B N M comma period slash"/>
      <row keys="space"/>
    </layer>
    <layer modifiers="shift caps">
      <row keys="tilde bang at hash dollar percent caret amp asterisk open-paren close-paren underscore plus"/>
      <row keys="q w e r t y u i o p open-curly close-curly pipe"/>
      <row keys="a s d f g h j k l colon double-quote"/>
      <row keys="z x c v b n m open-angle close-angle question"/>
      <row keys="space"/>
    </layer>
  </layers>
  <transforms type="simple">
    <transformGroup>
        <transform from="abc" to="Congratulations!" />
        <transform from="\m{y}z" to="a\m{b}" />
    </transformGroup>
    <transformGroup>
        <transform from="a\m{b}" to="b\m{c}" />
    </transformGroup>
  </transforms>
</keyboard3>
```

### CODE REVIEW

File: developer/src/tike/main/Keyman.System.KeymanCore.pas [hunk 1]
```diff
@@ -201,6 +205,35 @@   km_core_action_item = record
 
   pkm_core_action_item = ^km_core_action_item;
 
+  km_core_bool = uint32;
```
Comment (darcywong00) [inline developer/src/tike/main/Keyman.System.KeymanCore.pas:208]: Could this be an enum too?
https://github.com/keymanapp/keyman/blob/c494581117b76ac1453aa375faa9bbb99330cf86/core/include/keyman/keyman_core_api.h#L574

```suggestion
  km_core_bool = (
    KM_CORE_FALSE = 0,
    KM_CORE_TRUE = 1
  );
```

Comment (mcdurdin) [reply developer/src/tike/main/Keyman.System.KeymanCore.pas:208]: Unfortunately we can't really do that because then it becomes a uint8 internally.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_LDML_DEBUGGER (FAILED):** Tested with the attached PR build (Keyman Developer 17.0.247-alpha-test-10472) on Windows 10 OS and here is my observation: 1. Created an LDML keyboard project. 2. Opened the project. 3. Clicked the Keyboard / Test Keyboard option and got an error message. Seems to be an issue.

**Comment by mcdurdin** [comment]
@bharanidharanj the branch was not up to date so we didn't have the fix for that issue you experienced applied to this PR. I have updated the branch and am rebuilding now; please retest :smile:

@keymanapp-test-bot retest

**Comment by mcdurdin** [comment]
# Test Results

* **TEST_LDML_DEBUGGER (PASS):** all tests passed per the instructions given, running with Keyman Developer 17.0.248-alpha-test-10472 on a VM.

**Pull request merged**